ssl/ssl_sess.c

```diff
@@ -308,13 +308,91 @@ static int def_generate_session_id(const SSL *ssl, unsigned char *id,
     return 0;
 }
 
+int ssl_generate_session_id(SSL *s, SSL_SESSION *ss)
+{
+    unsigned int tmp;
+    GEN_SESSION_CB cb = def_generate_session_id;
+
+    switch (s->version) {
+    case SSL3_VERSION:
+    case TLS1_VERSION:
+    case TLS1_1_VERSION:
+    case TLS1_2_VERSION:
+    case TLS1_3_VERSION:
```

Comment:
IIRC, we still generate a session ID in this field to have a unique internal handle and simplify other code paths.  Someone (maybe Ilari or davidben?) pointed out security issues with the previous state of affairs w.r.t TLS 1.3